mii/deployment.py

```diff
@@ -66,86 +68,98 @@ def deploy(task,
         If deployment_type is `LOCAL`, returns just the name of the deployment that can be used to create a query handle using `mii.mii_query_handle(deployment_name)`
 
     """
+    if model_path is None and deployment_type == DeploymentType.LOCAL:
+        model_path = MII_MODEL_PATH_DEFAULT
+    elif model_path is None and deployment_type == DeploymentType.AML:
+        model_path = "model"
 
+    if not deployments and not all((model, task, deployment_name)):
+        assert deployment_tag is not None, "Deployment tag must be set when starting empty deployment"
+        create_score_file(deployment_tag=deployment_tag,
+                          deployment_type=deployment_type,
+                          deployments=None,
+                          model_path=model_path,
+                          port_map=None,
+                          lb_config=None)
+        return None
+
+    elif not deployments:
+        assert all((model, task, deployment_name)), "model, task, and deployment name must be set to deploy singular model"
+        deployments = [
+            DeploymentConfig(deployment_name=deployment_name,
+                             task=task,
+                             model=model,
+                             enable_deepspeed=enable_deepspeed,
+                             enable_zero=enable_zero,
+                             GPU_index_map=None,
+                             mii_config=mii.config.MIIConfig(**mii_config),
+                             ds_config=ds_config,
+                             version=version,
+                             deployed=False)
+        ]
+        deployment_tag = deployment_name
+    else:
+        assert deployment_tag is not None, "deployment_tag must be set to deploy multiple models"
     # parse and validate mii config
-    mii_config = mii.config.MIIConfig(**mii_config)
-    if enable_zero:
-        if ds_config.get("fp16", {}).get("enabled", False):
-            assert (mii_config.dtype == torch.half), "MII Config Error: MII dtype and ZeRO dtype must match"
-        else:
-            assert (mii_config.dtype == torch.float), "MII Config Error: MII dtype and ZeRO dtype must match"
-    assert not (enable_deepspeed and enable_zero), "MII Config Error: DeepSpeed and ZeRO cannot both be enabled, select only one"
+    for deployment in deployments:
+        mii_config = deployment.mii_config
+        if deployment.enable_zero:
+            if deployment.ds_config.get("fp16", {}).get("enabled", False):
+                assert (mii_config.dtype == torch.half), "MII Config Error: MII dtype and ZeRO dtype must match"
+            else:
+                assert (mii_config.dtype == torch.float), "MII Config Error: MII dtype and ZeRO dtype must match"
+        assert not (enable_deepspeed and enable_zero), "MII Config Error: DeepSpeed and ZeRO cannot both be enabled, select only one"
 
     # aml only allows certain characters for deployment names
     if deployment_type == DeploymentType.AML:
```

Comment:
Let's add an assert that prevents multi-model for AML deployments for the time being. We don't have a great way of testing this.